cave/reader/smac2_reader.py

```diff
@@ -0,0 +1,195 @@
+            raise FileNotFoundError("Specified format is \'SMAC2\', but no "
+                                    "\'paramstrings\'-file could be found "
+                                    "in %s" % self.folder)
+        configs_fn = os.path.join(self.folder, configs_fn.group())
+        self.logger.debug("Configurations loaded as kind-of csv from %s", configs_fn)
+        # Translate smac2 to csv
+        with open(rh_fn, 'r') as csv_file:
+            csv_data = list(csv.reader(csv_file, delimiter=',',
+                                       skipinitialspace=True))
+        header, csv_data = csv_data[0], np.array([csv_data[1:]])[0]
+        csv_data = pd.DataFrame(csv_data, columns=header)
+        csv_data = csv_data.apply(pd.to_numeric, errors='ignore')
+        self.logger.debug("Headers: " + str(list(csv_data.columns.values)))
+        data = pd.DataFrame()
+        data["config_id"] = csv_data["Run History Configuration ID"]
+        data["instance_id"] = csv_data["Instance ID"].apply(lambda x:
+                self.scen.train_insts[x-1])
+        data["seed"] = csv_data["Seed"]
+        data["time"] = csv_data["Runtime"]
+        if self.scen.run_obj == 'runtime':
+            data["cost"] = csv_data["Runtime"]
+        else:
+            data["cost"] = csv_data["Run Quality"]
+        data["status"] = csv_data["Run Result"]
+
+        # Load configurations
+        with open(configs_fn, 'r') as csv_file:
+            csv_data = list(csv.reader(csv_file, delimiter=',',
+                                       skipinitialspace=True))
+        configurations = {}  # id to config
+        for row in csv_data:
+            config_id = int(re.match(r'^(\d*):', row[0]).group(1))
+            params = [re.match(r'^\d*: (.*)', row[0]).group(1)]
+            params.extend(row[1:])
+            #self.logger.debug(params)
+            matches = [re.match(r'(.*)=\'(.*)\'', p) for p in params]
+            values = {m.group(1) : m.group(2) for m in matches}
+            rs = np.random.RandomState()
+            for name, value in values.items():
+                if isinstance(cs.get_hyperparameter(name), IntegerHyperparameter):
+                    values[name] = int(value)
+                if isinstance(cs.get_hyperparameter(name), FloatHyperparameter):
+                    values[name] = float(value)
+            configurations[config_id] = self._remove_inactive(cs, Configuration(cs,
+                                                    values=values,
+                                                    allow_inactive_with_values=True),
+                                                    values)
+        self.configurations = configurations
+
+        rh = CSV2RH().read_csv_to_rh(data,
+                                     pcs=cs,
+                                     configurations=configurations)
+
+        return (rh, validated_rh)
+
+    def get_trajectory(self, cs):
+        traj_fn = os.path.join(self.folder, '../traj-run-1.txt')
```

Comment:
Couldn't traj-run-1 also be called traj-run-5 or any number?
I think you should glob for traj-run-* in self.folder

Reply:
Right.